File: median-of-two-sorted-arrays.cpp
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        s=nums1+nums2
        s=sorted(s)
        l=len(s)
        if(l%2!=0):
            return s[l/2]
        else:
            return (float((s[(l-1)/2] +s[(l+1)/2])/2.0))

File: median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int totalLength = nums1.length + nums2.length;
        boolean isEven = (totalLength % 2 == 0);
        int midIndex1 = totalLength / 2;
        int midIndex2 = isEven ? midIndex1 - 1 : midIndex1;

        int i = 0, j = 0;
        int count = 0;
        int curr = 0, prev = 0;
        double ans = 0;
        while (count <= midIndex1) {
            prev = curr;
            if (i < nums1.length && (j >= nums2.length || nums1[i] <= nums2[j])) {
                curr = nums1[i++];
            } else {
                curr = nums2[j++];
            }
            count++;
        }

        if (isEven) {
            ans = (prev + curr) / 2.0;
        } else {
            ans = curr;
        }
        return ans;
    }
}
